zerver/lib/narrow.py

```diff
@@ -560,6 +579,8 @@ def by_dm(
         except (JsonableError, ValidationError):
             raise BadNarrowOperatorError("unknown user in " + str(operand))
 
+        self.check_not_both_stream_dm(is_dm_narrow=True)
+
```

Comment:
I think doing these more syntactic checks at the top rather than after doing some database queries to fetch data would feel more readable.